go/cmd/dolt/commands/sql.go

```diff
@@ -155,15 +169,46 @@ func (cmd SqlCmd) Exec(ctx context.Context, commandStr string, args []string, dE
 	saveMessage := apr.GetValueOrDefault(messageFlag, "")
 	saveName := apr.GetValueOrDefault(saveFlag, "")
 
-	err := validateSqlArgs(apr)
-	if err != nil {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+	_, forceBatchMode := apr.GetValue(batchFlag)
+
+	query, queryOK := apr.GetValue(queryFlag)
+	if !queryOK {
+		savedQueryName, exOk := apr.GetValue(executeFlag)
+
+		if exOk {
+			sq, err := dsqle.RetrieveFromQueryCatalog(ctx, root, savedQueryName)
+
+			if err != nil {
+				return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+			}
+
+			query = sq.Query
+			queryOK = true
```

Comment:
Don't like this kind of fallthrough logic, would much prefer extracting the logic used in the --query path into a method and invoking it here with an exit

Reply:
done